### PR TITLE
Global hook queries

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description of what you did:

First implementation of global hook queries to remove per plugin queries.

This branch includes:
- Groups are handeled everywhere (CM, plugins, core api)
- All query code is the same throughout the code => upadting in one place form now on.
- Implementation of unique constraints in sql databases and cleaner database migrations
- Convert all databae boolean values to real boolean (1 => true in sqlite) in the API responses

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [X] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [x] Framework
- [ ] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [x] MongoDB
- [x] MySQL
- [x] Postgres
- [x] SQLite

### CODE REVIEW

**Comment by derrickmehaffy** [comment]
@alexandrebodin Reading through the commit log, if I understand correctly this is moving the adminUI to use the normal services provided by the coreAPI (Controllers, Policies, Custom services, ect) correct? Instead of the admin having it's own set of those.

**Comment by alexandrebodin** [comment]
> @alexandrebodin Reading through the commit log, if I understand correctly this is moving the adminUI to use the normal services provided by the coreAPI (Controllers, Policies, Custom services, ect) correct? Instead of the admin having it's own set of those.

Not really. It's moving all the database logic to the hooks and generalize the strapi.query everywhere. This allows to handle stuff like groups, hooks etc in the future easily.

File: packages/strapi-hook-bookshelf/lib/queries.js [hunk 1]
```diff
@@ -1,24 +1,218 @@
...
+  const excludedKeys = assocKeys.concat(groupKeys);
+  // Returns an object without relational keys to persist in DB
+  const selectAttributes = values => {
+    return _.pickBy(values, (value, key) => {
+      return !excludedKeys.includes(key) && _.has(model.allAttributes, key);
+    });
   };
 
+  /**
+   * Find ne entry based on params
```
Comment (lauriejim) [inline packages/strapi-hook-bookshelf/lib/queries.js]: The `o` is missing.

File: packages/strapi-plugin-content-manager/test/index.test.e2e.js [hunk 2]
```diff
@@ -495,6 +495,7 @@ describe('Content Manager End to End', () => {
     });
 
     test('Get article2 with cat2', async () => {
+      data.articles;
```
Comment (lauriejim) [inline packages/strapi-plugin-content-manager/test/index.test.e2e.js]: Why this line?

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 3]
```diff
@@ -99,169 +122,199 @@ exports.connect = (provider, query) => {
...
-        if (err) {
-          callback(err);
-        } else {
-          callback(null, {
-            username: body.emails[0].value.split("@")[0],
-            email: body.emails[0].value
-          });
-        }
-      });
+      google
```
Comment (lauriejim) [inline packages/strapi-plugin-users-permissions/services/Providers.js]: Just a reminder, in an other PR I fixed the Google Auth provider - Please make sure to use the fix when you will merge with `master`

**Pull request merged**